### PR TITLE
feat: Timeout row count if it takes longer than 1/12 of frequency

### PR DESCRIPTION
**Pull request by tomasfarias**
## Problem

Turn row count into a best effort operation. This could have UI implications, but shouldn't impact _most_ users, as 1/12 of the frequency is enough time for most users, even those in 5 minute frequencies.

Row count in its current form is used for two reasons:
* Early return.
* UI metrics display.

I am considering completely phasing it out, but need to see how this would impact these two things first. Hence, why I opted for a timeout initially.

This should help cut down on time for those scenarios where count is slow, like when running unconstrained against a workflow event with significant and regular delays.
 

## Changes

Add a timeout to count in `start_batch_export_run`.

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

## Does this work well for both Cloud and self-hosted?

Yup.

## How did you test this code?

Covered by current tests.

### CODE REVIEW

File: posthog/temporal/batch_exports/batch_exports.py [hunk 1]
```diff
@@ -333,20 +334,33 @@ async def start_batch_export_run(inputs: StartBatchExportRunInputs) -> tuple[Bat
...
+            count = await asyncio.wait_for(
+                get_rows_count(
+                    client=client,
+                    team_id=inputs.team_id,
+                    interval_start=inputs.data_interval_start,
+                    interval_end=inputs.data_interval_end,
+                    exclude_events=inputs.exclude_events,
+                    include_events=inputs.include_events,
+                ),
+                timeout=(delta / 12).total_seconds(),
```
Comment (tomasfarias) [inline posthog/temporal/batch_exports/batch_exports.py:352]: Selected arbitrarily, it's the same factor as our jitter.

Comment (tomasfarias) [reply posthog/temporal/batch_exports/batch_exports.py:352]: Happy to take suggestions, otherwise sticking with vibes.

Comment (bretthoerner) [reply posthog/temporal/batch_exports/batch_exports.py:352]: Makes sense. It does seem a little wasteful to spend 25 seconds (if I'm doing my math right) of CH's time every 5 minutes doing a count that will fail/cancel anyway.

I wonder if unconstrained teams should skip count attempts? Or else we could have a whole other list to skip counts for. Anyway, not critical for now it's just a thought.

**Review by bretthoerner — APPROVED** [review]

**Comment by tomasfarias** [comment]
> Makes sense. It does seem a little wasteful to spend 25 seconds (if I'm doing my math right) of CH's time every 5 minutes doing a count that will fail/cancel anyway.

> I wonder if unconstrained teams should skip count attempts? Or else we could have a whole other list to skip counts for. Anyway, not critical for now it's just a thought.

I agree with you, ultimately I think we should get rid of counting here altogether. I am only holding on as unconstrained teams with relatively well sorted `_inserted_at` rows will still count very fast, so it would be a pity to mess up their metrics for no return in the meantime.

File: posthog/temporal/batch_exports/batch_exports.py (hunk 1 above)
Comment (tomasfarias) [reply posthog/temporal/batch_exports/batch_exports.py:352]: Ah, in a new comment. TLDR: I agree with you, just want to test out how this looks first, and eventually removing counts altogether.

**Pull request merged**